Terminal.Gui/Windows/FileDialog2.cs

```diff
@@ -0,0 +1,2118 @@
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using NStack;
+using Terminal.Gui.Trees;
+using static System.Environment;
+using static Terminal.Gui.OpenDialog;
+
+namespace Terminal.Gui {
+
+	/// <summary>
+	/// Modal dialog for selecting files/directories.  Has auto-complete and expandable
+	/// navigation pane (Recent, Root drives etc).
+	/// </summary>
+	public class FileDialog2 : Dialog {
```

Comment:
Before we merge this to v2 (I know it's still draft), we should replace the old (renaming this to `FileDialog`). Of course, this means ensuring all the existing Scenarios and Unit Tests that use the dialogs get updated/deleted/whateverneeded.